### PR TITLE
`src/doc`: Update `# needs`

### PR DESCRIPTION
**Pull request by mkoeppe**
Also some doctest cosmetics.

In part cherry-picked from #35095.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst [hunk 1]
```diff
@@ -199,11 +199,12 @@ It is also possible to define a polyhedron over algebraic numbers.
...
 ::
 
-    sage: sqrt_2 = AA(2)^(1/2)                                                # optional - sage.rings.number_field
-    sage: cbrt_2 = AA(2)^(1/3)                                                # optional - sage.rings.number_field
-    sage: timeit('Polyhedron(vertices = [[sqrt_2, 0], [0, cbrt_2]])')         # optional - sage.rings.number_field  # random
+    sage: # needs sage.rings.number_field
+    sage: sqrt_2 = AA(2)^(1/2)
+    sage: cbrt_2 = AA(2)^(1/3)
+    sage: timeit('Polyhedron(vertices = [[sqrt_2, 0], [0, cbrt_2]])')
     5 loops, best of 3: 43.2 ms per loop
```
Comment (kwankyu) [inline src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst:206]: `# random`?

File: src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst [hunk 2]
```diff
@@ -212,11 +213,12 @@ There is another way to create a polyhedron over algebraic numbers:
...
 
-    sage: K.<a> = NumberField(x^2 - 2, embedding=AA(2)**(1/2))                # optional - sage.rings.number_field
-    sage: L.<b> = NumberField(x^3 - 2, embedding=AA(2)**(1/3))                # optional - sage.rings.number_field
-    sage: timeit('Polyhedron(vertices = [[a, 0], [0, b]])')                   # optional - sage.rings.number_field  # random
+    sage: # needs sage.rings.number_field
+    sage: K.<a> = NumberField(x^2 - 2, embedding=AA(2)**(1/2))
+    sage: L.<b> = NumberField(x^3 - 2, embedding=AA(2)**(1/3))
+    sage: timeit('Polyhedron(vertices = [[a, 0], [0, b]])')
     5 loops, best of 3: 39.9 ms per loop
-    sage: P5 = Polyhedron(vertices = [[a, 0], [0, b]]); P5                    # optional - sage.rings.number_field
```
Comment (kwankyu) [inline src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst:219]: `# random`?

File: src/doc/en/thematic_tutorials/steenrod_algebra_modules.rst [hunk 3]
```diff
@@ -302,32 +309,32 @@ elements is a linear transformation::
...
-    [0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1]
-    [0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0]
-    [0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1]
-    [0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 1]
-    [0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 1]
-    [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1]
-    [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0]
-    [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1]
+    Domain:   Vector space quotient V/W of dimension 3 over Finite Field of size 2 where
+              V: Vector space of dimension 20 over Finite Field of size 2
```
Comment (kwankyu) [inline src/doc/en/thematic_tutorials/steenrod_algebra_modules.rst:313]: One space instead of three?

Comment (mkoeppe) [reply src/doc/en/thematic_tutorials/steenrod_algebra_modules.rst:313]: It's aligned with "Codomain" below

**Comment by kwankyu** [comment]
Otherwise, looks good to me.

File: src/doc/en/thematic_tutorials/steenrod_algebra_modules.rst (hunk 3 above)
Comment (kwankyu) [reply src/doc/en/thematic_tutorials/steenrod_algebra_modules.rst:313]: OK.

File: src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst (hunk 1 above)
Comment (kwankyu) [reply src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst:206]: Not fixed yet.

File: src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst [hunk 4]
```diff
@@ -199,11 +199,12 @@ It is also possible to define a polyhedron over algebraic numbers.
 
 ::
 
-    sage: sqrt_2 = AA(2)^(1/2)                                                # optional - sage.rings.number_field
-    sage: cbrt_2 = AA(2)^(1/3)                                                # optional - sage.rings.number_field
-    sage: timeit('Polyhedron(vertices = [[sqrt_2, 0], [0, cbrt_2]])')         # optional - sage.rings.number_field  # random
+    sage: # needs sage.rings.number_field
+    sage: sqrt_2 = AA(2)^(1/2)
+    sage: cbrt_2 = AA(2)^(1/3)
+    sage: timeit('Polyhedron(vertices=[[sqrt_2, 0], [0, cbrt_2]])')
```
Comment (dcoudert) [inline src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst]: You must put back the tag `# random`. The CI reports a doctest error:
```
sage -t --long --random-seed=286735480429121101562228604801325644303 doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst
**********************************************************************
File "doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst", line 205, in doc.en.thematic_tutorials.geometry.polyhedra_tutorial
Failed example:
    timeit('Polyhedron(vertices=[[sqrt_2, 0], [0, cbrt_2]])')
Expected:
    5 loops, best of 3: 43.2 ms per loop
Got:
    25 loops, best of 3: 21.9 ms per loop
**********************************************************************
1 item had failures:
   1 of 100 in doc.en.thematic_tutorials.geometry.polyhedra_tutorial
    [99 tests, 1 failure, 16.62 s]
```

**Review by dcoudert — CHANGES_REQUESTED** [review]

File: src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst (hunk 4 above)
Comment (mkoeppe) [reply src/doc/en/thematic_tutorials/geometry/polyhedra_tutorial.rst]: Thanks for catching this, fixed in f43552ef2b4

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thanks both!

**Pull request merged**